src/shogun/distance/MahalanobisDistance.cpp

```diff
@@ -39,20 +37,43 @@ bool CMahalanobisDistance::init(CFeatures* l, CFeatures* r)
 {
 	CRealDistance::init(l, r);
 
+	REQUIRE(
+	    lhs->get_feature_class() == C_DENSE,
+	    "Left hand side (was %s) has to be CDenseFeatures instance.\n",
+	    lhs->get_name());
```

Comment:
this segfaults if one of the features is not set (or does the above call check that?)

Reply:
Also: "Left hand side feature type (%s) has be be (%s)\n"
One should not use type names in the messages (think refactor), would be better to have a (static) method CDenseFeatures::get_name(). I'll send a patch and you can use that

Reply:
https://github.com/vinx13/shogun/blob/39e7dd2f390d13e665678be23b0b0221f1212033/src/shogun/distance/Distance.cpp#L80

Reply:
but the dense checks can be dropped as they are implicitly tested in `CRealDistance::init` and belove